File: app/src/main/java/com/example/zhibek_romanbekova_hw33_3m/FlowersAdapter.java
package com.example.zhibek_romanbekova_hw33_3m;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class FlowersAdapter extends RecyclerView.Adapter<FlowersViewHolder>{
    private ArrayList<String> flowersList;
    public FlowersAdapter(ArrayList<String> mealList) {
        this.flowersList = mealList;

    }
    @NonNull
    @Override
    public FlowersViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new FlowersViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_fowers,parent,false));
    }

    @Override
    public void onBindViewHolder(@NonNull FlowersViewHolder holder, int position) {
        holder.bind(flowersList.get(position));

    }

    @Override
    public int getItemCount() {
        return flowersList.size() ;
    }
}

File: app/src/main/java/com/example/zhibek_romanbekova_hw33_3m/FlowersAdapter.kt
package com.example.zhibek_romanbekova_hw33_3m

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.zhibek_romanbekova_hw33_3m.databinding.ItemFowersBinding

class FlowersAdapter(private val flowersList: ArrayList<Flowers>) :
    RecyclerView.Adapter<FlowersAdapter.FlowersViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FlowersViewHolder {
        return FlowersViewHolder(ItemFowersBinding.inflate(LayoutInflater.from(parent.context), parent, false))
    }

    override fun onBindViewHolder(holder: FlowersViewHolder, position: Int) {
        holder.bind(flowersList[position])
    }

    override fun getItemCount(): Int {
        return flowersList.size
    }

    inner class FlowersViewHolder(private val binding: ItemFowersBinding):ViewHolder(binding.root){
        fun bind(flowers: Flowers) {
            binding.listName.text = flowers.name
        }
    }
}